ttnn/cpp/ttnn/operations/data_movement/untilize_with_unpadding/untilize_with_unpadding.cpp

```diff
@@ -8,10 +8,53 @@
 #include "ttnn/common/constants.hpp"
 #include "ttnn/run_operation.hpp"
 
+#include "ttnn/operations/data_movement/common/common.hpp"
+#include "ttnn/operations/data_movement/reshape_view/reshape.hpp"
+
 using namespace tt::tt_metal;
 
+LegacyShape squeeze_output_shape(tt::tt_metal::LegacyShape output_shape) {
```

Comment:
I wonder if any generic method for arbitrary ND shape, as there will be requirement for higher dimension.

Reply:
@llongTT This looks generic to me. 

But @nardoTT I did have a question about this function. You write `std::vector<uint32_t> output_shape_4d(output_shape.rank());` so the output_shape_4d has the same rank as the original rank when passed to LegacyShape. Should this instead be a 4-element vector? It looks like the later elements are never filled in. or does UntilizeWithUnpadding's run have some special interpretation of 0's in output_end?

Reply:
It is actually a 4-element vector because the output_end_vector that we are passing to squeeze_output_shape function is 4D. squeeze_output_shape function just updates the value of the first element in output_end_vector to account for the dimensions greater than 4.